syft/frameworks/torch/hook/hook.py

```diff
@@ -492,6 +493,150 @@ def _hook_crypten(self):
         new_func = get_hooked_crypten_func("load", crypten_load)
         setattr(crypten, "load", new_func)
 
+    def _hook_crypten_module(self):
+        """Overloading crypten.nn.Module with PySyft functionality, the primary module
+           responsible for core ML functionality such as Neural network layers and
+           loss functions.
+           It is important to note that all the operations are actually in-place.
+        """
+        import crypten
+
+        def _check_encrypted(model):
+            if model.encrypted:
+                raise RuntimeError("Crypten model must be unencrypted to run PySyft operations")
+
+        crypten.nn.Module._check_encrypted = _check_encrypted
+
+        def module_is_missing_grad(model):
+            """Checks if all the parameters in the model have been assigned a gradient"""
+            for p in model.parameters():
+                if p.grad is None:
+                    return True
+            return False
+
+        def create_grad_objects(model):
```

Comment:
These functions are identical for torch & crypten, maybe we could factorize their usage?

Reply:
@LaRiffle I Agree ! would it make sense to put them in hook_module.py and import them to be used here?

Reply:
We agreed with George on refactoring all the functioning of crypten hooking to #3390